lib/restivus.coffee

```diff
@@ -295,12 +295,14 @@ class @Restivus
             searchQuery
           @userId = @user?._id
 
+        response = {status: 'success', data: auth}
+
         # Call the login hook with the authenticated user attached
-        loginResult = self._config.onLoggedIn.call(@)
-        if loginResult?
-          _.extend(auth, extra: loginResult)
+        extraData = self._config.onLoggedIn.call(this)
+        if extraData and _.isObject(extraData) and (not _.isEmpty(extraData))
```

Comment:
Why are we requiring the `extra` data be an object? A user may just want to return a string there; I won't judge :P I feel like we should try to avoid being opinionated here and just let the user return any defined value. If this was because of my unnecessary return of an empty object in the default `onLoggedIn` method, I apologize for that terrible code. 

Since the return value wasn't being used until very recently, and the empty object I return is more of a bug than a feature, I think it would be safe to change the default `onLoggedIn` and `onLoggedOut` functions to being `undefined`, do an existential check here like we had before, and add another to ensure the function isn't called unless the user overrides it in the API config. Putting that together should look something like:
```coffeescript
extraData = self._config.onLoggedIn?.call(this)
if extraData?
  _.extend(response.data, {extra: extraData})
```

And of course my useless `onLoggedIn` and `onLoggedOut` functions will need to be removed from the default config in the Restivus constructor (and the docs updated to reflect that those functions are undefined by default).

What do you think of that approach?